client/src/components/ChannelRouteSettings.tsx

```diff
@@ -30,76 +30,60 @@ class ChannelRouteSettings extends React.PureComponent<
-                        ' Channel ' +
-                        String(channel + 1) +
-                        ' to Fader ' +
-                        String(this.faderIndex + 1) +
-                        '?'
-                )
-            ) {
-                window.socketIoClient.emit(SOCKET_SET_ASSIGNED_FADER, {
-                    mixerIndex: mixerIndex,
-                    channel: channel,
-                    faderAssign: this.faderIndex,
+    handleAssignChannel(mixerIndex: number, channelIndex: number, event: any) {
+        console.log('Bind/Unbind Channel')
+        if (
+            window.confirm(
+                'Bind/Unbind Mixer ' +
+                String(mixerIndex + 1) +
+                ' Channel ' +
+                String(channelIndex + 1) +
+                ' from Fader ' +
+                String(this.faderIndex + 1)
+            )
+        ) {
+            // Check if channel already is assigned to another fader and remove that binding prior to bind it to the new fader
+            if (event.target.checked) {
+                this.props.fader.forEach((fader: IFader, index: number) => {
+                    if (fader.assignedChannels.some((assignedChan) => {
+                        return assignedChan.mixerIndex === mixerIndex && assignedChan.channelIndex === channelIndex
+                    })) {
+                        window.socketIoClient.emit(SOCKET_ASSIGN_CH_TO_FADER, {
+                            mixerIndex: mixerIndex,
+                            channel: channelIndex,
+                            faderIndex: index,
+                            assigned: false
+                        })
+                    }
                 })
             }
+
+
+            window.socketIoClient.emit(SOCKET_ASSIGN_CH_TO_FADER, {
+                mixerIndex: mixerIndex,
+                channel: channelIndex,
+                faderIndex: this.faderIndex,
+                assigned: event.target.checked
+            })
         }
     }
 
-    handleClearRouting() {
+    handleClearAllRouting() {
         if (window.confirm('REMOVE ALL FADER ASSIGNMENTS????')) {
-            this.props.chMixerConnections.forEach(
-                (chMixerConnection: IchMixerConnection, mixerIndex: number) => {
-                    chMixerConnection.channel.forEach(
-                        (channel: any, index: number) => {
-                            window.socketIoClient.emit(
-                                SOCKET_SET_ASSIGNED_FADER,
-                                {
-                                    mixerIndex: mixerIndex,
-                                    channel: index,
-                                    faderAssign: -1,
-                                }
-                            )
-                        }
-                    )
-                }
-            )
+            window.socketIoClient.emit(SOCKET_REMOVE_ALL_CH_ASSIGNMENTS)
         }
     }
 
-    handle11Routing() {
+    handle1to1Routing() {

```

Comment:
A suggestion regarding the casing could be to rename this to `handle1To1Routing` or `handleOneToOneRouting`. A reason for favor the latter is that the `1` might be confused with `l` or `I` depending on the font/theme used in an IDE.